android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerModule.kt

```diff
@@ -146,59 +146,73 @@ class RNGestureHandlerModule(reactContext: ReactApplicationContext?) : ReactCont
         hasCustomActivationCriteria = true
+        minDistFromActivationCriteria = Float.MIN_VALUE
       }
       if (config.hasKey(KEY_PAN_FAIL_OFFSET_RANGE_Y_END)) {
         handler.setFailOffsetYEnd(PixelUtil.toPixelFromDIP(config.getDouble(KEY_PAN_FAIL_OFFSET_RANGE_Y_END)))
         hasCustomActivationCriteria = true
+        minDistFromActivationCriteria = Float.MIN_VALUE
+      }
+
+      if (config.hasKey(KEY_PAN_ACTIVE_OFFSET_X_START)) {
+        handler.setActiveOffsetXStart(PixelUtil.toPixelFromDIP(config.getDouble(KEY_PAN_ACTIVE_OFFSET_X_START)))
+        hasCustomActivationCriteria = true
+        minDistFromActivationCriteria = Float.MAX_VALUE
       }
+      if (config.hasKey(KEY_PAN_ACTIVE_OFFSET_X_END)) {
+        handler.setActiveOffsetXEnd(PixelUtil.toPixelFromDIP(config.getDouble(KEY_PAN_ACTIVE_OFFSET_X_END)))
+        hasCustomActivationCriteria = true
+        minDistFromActivationCriteria = Float.MAX_VALUE
+      }
+      if (config.hasKey(KEY_PAN_ACTIVE_OFFSET_Y_START)) {
+        handler.setActiveOffsetYStart(PixelUtil.toPixelFromDIP(config.getDouble(KEY_PAN_ACTIVE_OFFSET_Y_START)))
+        hasCustomActivationCriteria = true
+        minDistFromActivationCriteria = Float.MAX_VALUE
+      }
+      if (config.hasKey(KEY_PAN_ACTIVE_OFFSET_Y_END)) {
+        handler.setActiveOffsetYEnd(PixelUtil.toPixelFromDIP(config.getDouble(KEY_PAN_ACTIVE_OFFSET_Y_END)))
+        hasCustomActivationCriteria = true
+        minDistFromActivationCriteria = Float.MAX_VALUE
+      }
+
       if (config.hasKey(KEY_PAN_MIN_VELOCITY)) {
         // This value is actually in DPs/ms, but we can use the same function as for converting
         // from DPs to pixels as the unit we're converting is in the numerator
         handler.setMinVelocity(PixelUtil.toPixelFromDIP(config.getDouble(KEY_PAN_MIN_VELOCITY)))
         hasCustomActivationCriteria = true
+        minDistFromActivationCriteria = Float.MAX_VALUE
       }
       if (config.hasKey(KEY_PAN_MIN_VELOCITY_X)) {
         handler.setMinVelocityX(PixelUtil.toPixelFromDIP(config.getDouble(KEY_PAN_MIN_VELOCITY_X)))
         hasCustomActivationCriteria = true
+        minDistFromActivationCriteria = Float.MAX_VALUE
       }
       if (config.hasKey(KEY_PAN_MIN_VELOCITY_Y)) {
         handler.setMinVelocityY(PixelUtil.toPixelFromDIP(config.getDouble(KEY_PAN_MIN_VELOCITY_Y)))
         hasCustomActivationCriteria = true
+        minDistFromActivationCriteria = Float.MAX_VALUE
       }
 
       // PanGestureHandler sets minDist by default, if there are custom criteria specified we want
       // to reset that setting and use provided criteria instead.
       if (config.hasKey(KEY_PAN_MIN_DIST)) {
         handler.setMinDist(PixelUtil.toPixelFromDIP(config.getDouble(KEY_PAN_MIN_DIST)))
       } else if (hasCustomActivationCriteria) {
-        handler.setMinDist(Float.MIN_VALUE)
+        handler.setMinDist(minDistFromActivationCriteria)
```

Comment:
As discussed I think it'd be ok to just throw here and also add a more descriptive message on the JS side when someone attempts to use minDist and activeOffset/failOffset together. As it'd be difficult to come up with a intuitive way of doing that it'd be safer to just disallow it completely as I think all the possible usecases here can be build using some other set of attributes.

Reply:
I added errors in the JS in https://github.com/software-mansion/react-native-gesture-handler/pull/1696/commits/41ccfc0797a13db1c7c4d0f9bf79e471e0745cb2, but I think we cannot throw in Android. Setting `minDist` to `Float.MAX_VALUE` here makes the handler not activate until `activeOffsetX/Y` condition is met. I'd say that showing error is ok for now and we can revisit the issue in the future if there are still problems with it. What do you think?

Reply:
console.error is actually an equivalent to throw in production. That is this error would result in an app crash in production.

I don't quite understand the reason why we can't throw?